includes/Modules/Idea_Hub.php

```diff
@@ -384,15 +328,38 @@ protected function create_data_request( Data_Request $data ) {
 					);
 				};
 			case 'GET:saved-ideas':
-				// @TODO: Implement this with the real API endpoint.
-				return function() {
-					return array();
-				};
+				return $this->fetch_ideas( 'saved' );
 			case 'POST:update-idea-state':
-				// @TODO implementation
-				return function() {
-					return null;
-				};
+				if ( ! isset( $data['name'] ) ) {
+					return new WP_Error(
+						'missing_required_param',
+						/* translators: %s: Missing parameter name */
+						sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'name' ),
+						array( 'status' => 400 )
+					);
+				}
+
+				$parent = $this->get_parent_slug();
+				$body   = new Google_Service_Ideahub_GoogleSearchIdeahubV1alphaIdeaState();
+
+				$body->setName( $data['name'] );
+
+				if ( isset( $data['saved'] ) ) {
+					$parent = $parent . '/ideaStates/saved';
+					$body->setSaved( filter_var( $data['saved'], FILTER_VALIDATE_BOOL ) );
+				} elseif ( isset( $data['dismissed'] ) ) {
```

Comment:
This shouldn't/doesn't need to be an either/or. It is acceptable to specify both states at the same time, i.e. it would be more appropriate to have two individual `if` clauses here. Only if none of the two is provided, the request is pointless, so returning an error then makes sense.

Reply:
Updated.